xtra-libp2p-offer/src/lib.rs

```diff
@@ -1,9 +1,158 @@
+use model::olivia::BitMexPriceEventId;
+use model::ContractSymbol;
+use model::FundingRate;
+use model::Leverage;
+use model::OfferId;
+use model::OpeningFee;
+use model::Origin;
+use model::Position;
+use model::Price;
+use model::Timestamp;
+use model::TxFeeRate;
+use model::Usd;
+use serde::Deserialize;
+use serde::Serialize;
+use std::fmt;
+use time::Duration;
+
 pub mod maker;
 mod protocol;
 pub mod taker;
 
 pub const PROTOCOL: &str = "/itchysats/offer/1.0.0";
 
+#[derive(Clone, Serialize, Deserialize, PartialEq)]
+pub struct MakerOffers {
+    pub long: Option<Offer>,
+    pub short: Option<Offer>,
+    pub tx_fee_rate: TxFeeRate,
+    pub funding_rate_long: FundingRate,
+    pub funding_rate_short: FundingRate,
+}
```

Comment:
👍 I like the introduction of the new type, but I would change the shape as mentioned in a recent comment.

Reply:
I think we should also change the shape of `model::MakerOffers` though. The original design has some problems as discussed.